Differential datalog CI build fails

@qishen it seems the build is currently broken due to the `use of unstable library feature`. (Previously, this was masked by another CI bug in the build.)

https://github.com/TransformationToolContest/ttc2018liveContest/runs/6339663167?check_suite_focus=true#step:6:1386

```console
error[E0658]: use of unstable library feature 'available_parallelism'
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/dashmap-5.3.3/src/lib.rs:55:6
   |
55 |     (std::thread::available_parallelism().map_or(1, usize::from) * 4).next_power_of_two()
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #74479 <https://github.com/rust-lang/rust/issues/74479> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `dashmap` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
Running benchmark with root directory /ttc
Building differential
Building differential-08
Building differential-datalog
Traceback (most recent call last):
  File "scripts/run.py", line 215, in <module>
    build(config, args.skip_tests and not args.test)
  File "scripts/run.py", line 41, in build
    subprocess.check_call(config.get('build', 'default'), shell=True)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'ddlog/download.sh && DIFFERENTIAL_DATALOG_DISABLE_DEBUG= ./build.sh snq1 && DIFFERENTIAL_DATALOG_DISABLE_DEBUG= ./build.sh snq2' returned non-zero exit status 101.
The command '/bin/sh -c scripts/run.py -b' returned a non-zero code: 1
```

Could you please suggest a way / submit a PR to fix this? Do we have to use an unstable Rust version in the CI?

--- a/docker/Dockerfile-differential
+++ b/docker/Dockerfile-differential
@@ -1,7 +1,7 @@
 FROM ftsrg/ttc2018:common
 
-RUN apt-get update && \
-    apt-get install -yq --no-install-recommends rustc cargo \
+RUN apt-get update \
+    && apt-get install -yq --no-install-recommends gcc libc-dev\
     && rm -rf /var/lib/apt/lists/* \
     # install sccache
     && wget -qO sccache.tar.gz "https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz" \
@@ -11,7 +11,31 @@ RUN apt-get update && \
     && cd sccache-* \
     && chmod u+x sccache \
     && mv sccache /usr/local/bin/
+    
 ENV RUSTC_WRAPPER=sccache
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH
+
+# Install a specific nightly version of Rust toolchain
+RUN set -eux; \
+    dpkgArch="$(dpkg --print-architecture)"; \
+    case "${dpkgArch##*-}" in \
+        amd64) rustArch='x86_64-unknown-linux-gnu' ;; \
+        arm64) rustArch='aarch64-unknown-linux-gnu' ;; \
+        *) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
+    esac; \
+    \
+    url="https://static.rust-lang.org/rustup/dist/${rustArch}/rustup-init"; \
+    wget "$url"; \
+    chmod +x rustup-init; \
+    ./rustup-init -y --no-modify-path --default-toolchain nightly-2022-05-18; \
+    rm rustup-init; \
+    chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
+    rustup --version; \
+    cargo --version; \
+    rustc --version;
+
 
 # small and stable files
 COPY expected-results expected-results

